File: hashing/counting_elements.js
const countElements = (arr) => {
    const numSet = new Set();
    for (const x of arr)
    {
        numSet.add(x);
    }

    let count = 0;
    for (const x of arr)
    {
        if (numSet.has(x + 1))
        {
            count++;
        }
    }

    return count;
}

File: hashing/counting_elements.java
public int countElements(int[] arr)
{
    Set<Integer> numSet = new HashSet<>();
    for (int x : arr)
    {
        numSet.add(x);
    }

    int count = 0;
    for (int x : arr)
    {
        if (numSet.contains(x + 1))
        {
            count++;
        }
    }

    return count;
}